src/program/program.service.ts

```diff
@@ -1083,9 +1083,15 @@ export class ProgramService {
 			data: {
 				...(type && { type }),
 				...(link && { link }),
-				parent: parent ? { connect: { id: parent } } : undefined
+				parent: parent ? { connect: { id: parent } } : undefined,
+				primary: primary
 			}
 		});
+		// updatecontent function implemented above is accepting only certain ID'S Content data but not Array of Content
+		// Accordingly I have implemented
+		// checked for primary source and setting it to secondary resource
+
+		if (data.data.primary === true) data.data.primary = false;
```

Comment:
- How will the service respond if we want to set an already primary content to secondary (passing in primary = false)?
- Will the service fail if we update a content to be primary and there is already a primary content available?
- What happens if we delete the only primary content in the array? Will the service update the first content in the array to be the primary automatically?

Please rethink your logic here, because this isn't adhering our control flows discussed previously. As mentioned in the ticket, you might want to offload this logic to the resolver and run the update service in different scenarios to make sure our requirements are met.